upgrade.py

```diff
@@ -503,6 +531,24 @@ def completeUpgrade(self, mounts, prev_install, target_disk, backup_partnum, log
         print("UPGRADE=true", file=state)
         state.close()
 
+        interface_rename_dir = os.path.join(
+            mounts['root'],
+            'etc/sysconfig/network-scripts/interface-rename-data/'
+        )
+        src_file_path = os.path.join(interface_rename_dir, 'dynamic-rules.json')
+        dest_file_path = os.path.join(
+            mounts['root'],
+            constants.FIRSTBOOT_DATA_DIR,
+            'initial_network_device_rules.conf'
+        )
+        if os.path.exists(src_file_path):
+            if self.convert_interface_rename_data(src_file_path, dest_file_path):
+                logger.log(f'Removing {interface_rename_dir} ...')
+                shutil.rmtree(interface_rename_dir)
```

Comment:
Can't you read the data directly from the backup instead of copying it to the new installation and then immediately removing it?